### DISCUSSION TITLE
Can someone please share an example of how to use @flow with typescript?

### DISCUSSION THREAD

**Discussion by suparngp** [mobxjs/mobx#2556] [Q&A] [answered]
I am having a bit of a trouble with `@flow`. I was using `flow()` wrapper with class properties like `createUser = flow(*function()...)`. Now that `@flow` is available, I was hoping to convert everything to 
```
@ flow
*createUser(){}
```
But now, that means I have to specify the correct return type of the function, or else typescript complains. So I tried something like this

```
@ flow
*createUser(): Generator<unknown, void, Result> {}
```
and wrapping the using it as following:

`const result = await flowResult(instance.createUser(....));`
However, `result` type is `void` (cuz I wrote it like that in the return type) and I don't know how to make it return a `CancellablePromise`.

**Comment by kubk** [comment, marked as answer]
There are a lot of problems with generators in TypeScript. Basically, TypeScript can't infer the type of yield expressions in such simple cases:
```js
const foo = yield 1; // variable foo has type 'any' instead of 'number' 
```
Here is the open issue: https://github.com/microsoft/TypeScript/issues/32523
Currently, generators in TypeScript aren't type-safe and not very convenient to use. Mobx can't fix that.
